apps/s_server.c

```diff
@@ -2747,12 +2747,14 @@ static int init_ssl_connection(SSL *con)
                 wbio = SSL_get_wbio(con);
                 if (wbio) {
                     BIO_get_fd(wbio, &fd);
-                }
-
-                if (!wbio || BIO_connect(fd, client, 0) == 0) {
-                    BIO_printf(bio_err, "ERROR - unable to connect\n");
-                    BIO_ADDR_free(client);
-                    return 0;
+                } else {
+                    if (BIO_connect(fd, client, 0) == 0) {
```

Comment:
This change makes no sense. `fd` is -1 here, so the `BIO_connect` call is always going to fail? Also the logic before immediately errored out if `wbio == NULL`, but here you only attempt the connect if it is NULL (which shouldn't normally happen) - so we never normally hit this whole `else` branch.

Reply:
i literally have no idea what i was thinking; sorry.